db-service/lib/cqn2sql.js

```diff
@@ -862,8 +865,8 @@ class CQN2SQLRenderer {
       if (converter && sql[0] !== '$') sql = converter(sql, element)
 
       let val = _managed[element[annotation]?.['=']]
+        || !isUpdate && (element['@cds.valid.from'] ? '$valid.from' : element['@cds.valid.to'] ? '$valid.to' : null)
```

Comment:
I think we did not support this so far as we cannot achieve this on SAP HANA. We do an `UPSERT... WITH PRIMARY KEY` and do not list the specific columns.

Reply:
@johannes-vogel I am not 100% certain what your concern is.

The current HANA UPSERT implementation creates a join condition. [here](https://github.com/cap-js/cds-dbs/blob/main/hana/lib/HANAService.js#L641)

Which should look something like:
```SQL
UPSERT "TABLE" (
  A,B,C
) SELECT
  NEW.A,
  COALESCE(NEW.B,OLD.B) AS B,
  CASE OLD.A IS NULL THEN NEW.C ELSE OLD.C END
FROM (SELECT ... FROM JSON_TABLE(...))
LEFT JOIN "TABLE" AS OLD ON NEW.A=OLD.A
```

Reply:
I was referring to the old db implementation but the concern is invalid anyway as the `@cds.valid.from` column is marked as table key anyway.